test/test_lazy.py

```diff
@@ -0,0 +1,202 @@
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+import numpy as np
+from functools import partial
+import pyopencl as cl
+import pyopencl.tools as cl_tools
+import pyopencl.array as cla  # noqa
+import pyopencl.clmath as clmath  # noqa
+from meshmode.array_context import (  # noqa
+    PyOpenCLArrayContext,
+    PytatoPyOpenCLArrayContext
+)
+from meshmode.dof_array import thaw
+
+from meshmode.array_context import (  # noqa
+    pytest_generate_tests_for_pyopencl_array_context
+    as pytest_generate_tests)
+
+import pytest  # noqa
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def _op_test_fixture(cl_ctx):
+    queue = cl.CommandQueue(cl_ctx)
+    actx = PytatoPyOpenCLArrayContext(
+        queue,
+        allocator=cl_tools.MemoryPool(cl_tools.ImmediateAllocator(queue)))
+
+    n = 4
+
+    from meshmode.mesh.generation import generate_regular_rect_mesh
+    mesh = generate_regular_rect_mesh(
+        a=(-0.5,)*2,
+        b=(0.5,)*2,
+        nelements_per_axis=(n,)*2,
+        boundary_tag_to_face={
+            "x": ["-x", "+x"],
+            "y": ["-y", "+y"]
+        })
+
+    from grudge.eager import EagerDGDiscretization
+    discr = EagerDGDiscretization(actx, mesh, order=2)
+
+    return actx, discr
+
+
+def _rel_linf_error(discr, x, y):
+    from mirgecom.fluid import ConservedVars
+    if isinstance(x, ConservedVars):
+        return _rel_linf_error(discr, x.join(), y)
+    if isinstance(y, ConservedVars):
+        return _rel_linf_error(discr, x, y.join())
+    return discr.norm(x - y, np.inf) / discr.norm(y, np.inf)
+
+
+# FIXME: Re-enable and fix up if/when standalone gradient operator exists
+# def test_lazy_op_gradient(ctx_factory):
+#     cl_ctx = ctx_factory()
+#     actx, discr = _op_test_fixture(cl_ctx)
+#
+#     from grudge.dof_desc import DTAG_BOUNDARY, DISCR_TAG_BASE
+#     from mirgecom.diffusion import (
+#         _gradient_operator,
+#         DirichletDiffusionBoundary,
+#         NeumannDiffusionBoundary)
+#
+#     boundaries = {
+#         DTAG_BOUNDARY("x"): DirichletDiffusionBoundary(0.),
+#         DTAG_BOUNDARY("y"): NeumannDiffusionBoundary(0.)
+#     }
+#
+#     def op(alpha, u):
+#         return _gradient_operator(
+#             discr, DISCR_TAG_BASE, alpha, boundaries, u)
+
+#     compiled_op = actx.compile(op)
+#     alpha = discr.zeros(actx) + 1
+#     u = discr.zeros(actx)
+#     compiled_op(alpha, u)
+
+
+# FIXME: Re-enable and fix up if/when standalone divergence operator exists
```

Comment:
Would this one work?
https://github.com/illinois-ceesd/mirgecom/blob/aad1a100f4cf836c3ffce2a9c3c964e9844b7d61/mirgecom/operators.py#L31

Reply:
👍 Added.